cmd/bb_storage/BUILD.bazel

```diff
@@ -30,10 +37,12 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
-container_image(
+go_image(
     name = "bb_storage_container",
-    entrypoint = ["/bb_storage"],
-    files = [":bb_storage"],
+    embed = [":go_default_library"],
+    goarch = "amd64",
```

Comment:
Is this really required? According to [documentation](https://github.com/bazelbuild/rules_docker#go_image) both `goarch` and `goos` can be omitted it seems.

Reply:
it is required if building machine is not linux/amd64. This is required to allow to build docker images on mac machine.

Reply:
You could still achieve this using `bazel build --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 //...`, although I'm not sure which is best. If bb-storage were to be ported on more platforms, setting the default to linux_amd64 might not be the best choice anymore. @EdSchouten, thoughts?

Reply:
I agree with @mickael-carl here. Even though the container images are most likely to be used on linux/amd64, it's not unrealistic for people to also build them for linux/arm64.

Reply:
You still can build containers for the full list of supported architectures using this way:
```
[ go_image(
  name = 'image_' + OS + ' '+ ARCH,
  embed = [":go_default_library"],
  goarch = ARCH,
  goos = OS,
  pure = 'on',
  visibility = ["//visibility:public"],
)  for (ARCH, OS) in [
    ('amd64', 'linux'),
    ('amd64', 'freebsd'),
    ('armv7', 'linux'),
    # add your favorite architecture/os here
  ]
```
Of course you can do this using platforms=.... flag. The big difference, with goarch in BUILD file you DON'T need to specify any flags and still build a any configured architecture.
so instead of 
`bazel build --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 //...`
for linux, then 10 other platforms for other architectures, you just do
`bazel build //...`
and build all your configured images for all architectures in one single pass. Moreover, you can push all your containers into a registry using single runnable command.
This is exactly what bazel is moving to as explained in its roadmap here https://bazel.build/roadmaps/configuration.html
the roadmap goal is to make `$ bazel build :all` **just works, for whatever platform(s) you build for** and we, happy users of go and bazel, are fortunate to enjoy it working right now.

Reply:
let me know if you guys still think you want to stick to command line flags.

Reply:
ping?

Reply:
I do see a problem with this approach, namely that this will force us to maintain a list of `(ARCH, OS)` somewhere, if not in multiple places, across several repositories. This will likely end up being quite tedious and error-prone.
Currently docker supports [multi-arch containers](https://docs.docker.com/engine/reference/commandline/manifest/), but this hasn't really been picked up by `rules_docker` at this time. It is however [considered](https://github.com/bazelbuild/rules_docker/issues/300).
Furthermore, you are right that `bazel build :all` should just work, and it might just already, *but*, it is not as efficient currently as what [the design document](https://docs.google.com/document/d/1vc8v-kXjvgZOdQdnxPTaV0rrLxtP2XwnD2tAZlYJOqw/) proposes. Eventually when this lands, you will simply be able to configure a list of targets through a rule, that you will be able to pass to Bazel as build configuration, which is already much cleaner imho.
In the mean time, I think it is best if we stick to command-line flags though, just to avoid adding hard-coded values in build files. You could also specify those platforms [as config in your bazelrc](https://github.com/kubernetes/kubernetes/blob/master/build/root/.bazelrc#L29) as well, for convenience.

Reply:
rolled back

Reply:
I think this wasn't reverted in the end? Could you do so as well? Then I can merge it finally!

Reply:
oops sorry for that. it was only partially reverted. fixed now.